site/plugins.md

```diff
@@ -390,8 +391,234 @@ func (p *Plugin) Report(w http.ResponseWriter, r *http.Request) {
+
+The topology structure consists of the following attributes:
+
+- `nodes` - is the list of the nodes that compose the topology.
+- `controls` - contains the description of the controls that apply to the specific topology.
+- `metadata_templates` - contains the templates used to render data into the Scope UI.
+- `table_templates` - contains the templates used to render tables into the Scope UI.
+
+**Note**: These attribute are not required. But a topology with no `nodes` does not have any information to render, `metadata_templates`, as well as `table_templates`, are needed to know how to render the information carried by `nodes` in the Scope UI.
+
+### Nodes
+A Node contains information about a specific element of a topology.
+For example the Host topology will contain nodes describing all the hosts in it.
+The same is for containers and Container topology, pods and Pod topology and so on.
+Nodes are represented as follows:
+
+```json
+{
+  "Host": {
+    "nodes" : {
+      "hostID;<host>": {...}
+    }
+  },
+  "Container": {
+    "nodes" : {
+      "containerID;<container>": {...}
+    }
+  },
+  ...
+}
+```
+
+Nodes are stored in dictionary. The ID of nodes representing hosts or containers have the format `ID;<type>`, with type equal `host` or `container` accordingly.
+
+A node contains all the information about the represented object (e.g. host, container, pod, etc.).
+In particular a node may contain:
+
+- `latest` - the latest values to display.
+- `latestControls` - the latest available controls.
+- `metrics` - the collection of metrics to display in the UI.
+
+### Controls
+Controls describe interfaces that expose actions that the user can perform on different objects (e.g. host, container, etc.).
+Controls are an element of nodes. In this way, each control in a node is attached to it and performs an action on the object described by the node itself. Below is an example of how controls are represented in the JSON report.
+
+```json
+"latestControls": {
+	"switchToIOWait": {
+		"timestamp": "2016-11-17T08:53:03.171438309Z",
+		"value": {
+			"dead": false
+		}
+	},
+	"switchToIdle": {
+		"timestamp": "2016-11-17T08:53:03.171438309Z",
+		"value": {
+			"dead": true
+		}
+	}
+}
+``` 
+
+- `timestamp` is the timestamp of when the control was exposed.
+- `value` is an object containing the control value. At the moment, only the state is available.
+ - `Dead` is a boolean to know the state (active, dead) of a control. It is useful to show controls only when they are in a usable state.
+
+### Metadata Templates
+Metadata Templates describe of a particular metadata.
+This description is used to extract metadata from a node and display it on Scope UI.
+
+```json
+"metadata_templates": {
+      "metadata-template-id": {
+        "id": "metadata-template-id",
+        "label": "Human-readable description",
+        "priority": 13.5,
+        "from": "latest"
+      },
+      "another-metadata-template-id": {...}
+}
+```
+
```

Comment:
There's no mention of the `dataType` value here, which is vital for displaying non-string values.
What are the possible values for dataType, and how should the (string?) values in `latest` be formatted so that the formatting behaves correctly?